Automattic/wp-calypso
# Help Center: use URL params for determining editor canvas state

Partially addresses #95960 

## Proposed Changes

* Use the `@wordpress/router` package to determine the canvas state instead of `getCanvasMode` which will be removed in GB 19.6.0

https://github.com/user-attachments/assets/dee676e8-96fd-4672-befb-7c7dbdd38a6e

## Why are these changes being made?


* When GB 19.6.0 is installed and active, errors are thrown in the UI and the Help Center widget is not loaded

## Testing Instructions



* Activate GB 19.6.0-rc.2 on your sandbox
* cd into apps/help-center
* Run yarn dev --sync
* Sandbox your site and widgets.wp.com
* Go to the Site Editor (Appearance -> Editor)
* Make sure help icon displays
* Make sure you can open/close the help center by clicking the icon
* Click into the editor canvas
* Make sure no UI errors related to help center appear
* Make sure the help center icon is present
* Go to the Post Editor
* Make sure the help center icon is there and is functioning
* Make sure console errors are resolved

## Pre-merge Checklist
